### ISSUE TITLE
MAME: running with -debug option doesn't cleanup on exit

### ISSUE THREAD

**Issue by angelosa** [haikuports/haikuports#10402]
Steps to reproduce:
1. mame <whatever_romset_id> -debug
2. Terminal outputs `QObject::connect: Cannot connect (nullptr)::applicationQuit() to QHaikuIntegration::platformAppQuit()`
3. On close (issue an exit from debugger prompt), process will stay resident, requiring a kill process.

**Comment by pulkomandy** [comment]
I can confirm this, possibly an interaction between SDL and Qt, as MAME uses both in this case?

**Comment by angelosa** [comment]
Yes, MAME uses SDL for main window vs. QT for debugger, cfr.
https://docs.mamedev.org/commandline/commandline-all.html#core-video-options
https://docs.mamedev.org/commandline/commandline-all.html#mame-commandline-debugger

**Comment by PocketNerdIO** [comment]
This issue no longer occurs in 0.268, possibly as a result of this commit: https://github.com/mamedev/mame/commit/d32f7e3a99a0dd508ecdcfe4495451cfc5458ed2

A very similar error message still appears, but MAME exits cleanly.

```
TestQObject::connect: Cannot connect (nullptr)::applicationQuit() to QHaikuIntegration::platformAppQuit()
```

I tested this with `SUBSET=psion/psion3a.cpp`. I'm currently waiting for the full app to compile, without a `SUBSET`. Once this is finished and I've checked that it's working, I'll make a pull request for 0.268.
